cvmfs/s3fanout.cc

```diff
@@ -816,16 +875,18 @@ Failures S3FanoutManager::InitializeRequest(JobInfo *info, CURL *handle) const {
     assert(retval == CURLE_OK);
     retval = curl_easy_setopt(handle, CURLOPT_NOBODY, 1);
     assert(retval == CURLE_OK);
-    info->http_headers =
-      curl_slist_append(info->http_headers, "Content-Length: 0");
 
-    if (info->request == JobInfo::kReqDelete) {
+    if ((info->request == JobInfo::kReqDelete) ||
+        (info->request == JobInfo::kReqHeadPut))
+      {
       retval = curl_easy_setopt(handle, CURLOPT_CUSTOMREQUEST,
                                 GetRequestString(*info).c_str());
       assert(retval == CURLE_OK);
     } else {
       retval = curl_easy_setopt(handle, CURLOPT_CUSTOMREQUEST, NULL);
       assert(retval == CURLE_OK);
+      info->http_headers =
```

Comment:
I think we can remove this, too.  We'd then simply not set a content-length for DELETE and HEAD requests.

Reply:
done